conda_smithy/linter/hints.py

```diff
@@ -291,24 +292,67 @@ def hint_noarch_python_use_python_min(
                 ):
                     break
             else:
-                hint += (
+                hint.add(
                     f"\n   - For the `{section_name}` section of the recipe, you should usually use `{report_syntax}` "
                     f"for the `python` entry."
                 )
+    return hint
 
-        if hint:
-            hint = (
-                (
-                    "`noarch: python` recipes should usually follow the syntax in "
-                    "our [documentation](https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python) "
-                    "for specifying the Python version."
-                )
-                + hint
-                + (
-                    "\n   - If the package requires a newer Python version than the currently supported minimum "
-                    "version on `conda-forge`, you can override the `python_min` variable by adding a "
-                    "Jinja2 `set` statement at the top of your recipe (or using an equivalent `context` "
-                    "variable for v1 recipes)."
+
+def hint_noarch_python_use_python_min(
+    host_reqs,
+    run_reqs,
+    test_reqs,
+    outputs_section,
+    noarch_value,
+    recipe_version,
+    hints,
+):
+    hint = set()
+
+    if outputs_section:
+        for output in outputs_section:
+            requirements = output.get("requirements", {})
+            if isinstance(requirements, Mapping):
+                output_host_reqs = requirements.get("host")
+                output_run_reqs = requirements.get("run")
+            else:
+                output_host_reqs = requirements
+                output_run_reqs = requirements
+
+            hint.update(
+                hint_noarch_python_use_python_min_inner(
+                    output_host_reqs or [],
+                    output_run_reqs or [],
+                    get_all_test_requirements(output, [], recipe_version),
+                    output.get("build", {}).get("noarch"),
+                    recipe_version,
                 )
             )
-            hints.append(hint)
+    else:
+        hint.update(
+            hint_noarch_python_use_python_min_inner(
+                host_reqs,
+                run_reqs,
+                test_reqs,
+                noarch_value,
+                recipe_version,
+            )
+        )
+
+    if hint:
+        hint = (
+            (
+                "`noarch: python` recipes should usually follow the syntax in "
+                "our [documentation](https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python) "
+                "for specifying the Python version."
+            )
+            + "".join(sorted(hint))
```

Comment:
This sorts the entire set of hints based on the full hint. Don't we only want to use the name of the output?

Reply:
We don't include the output name right now. We just give a list of "generic" hints that apply to at least one output in the recipe. Do you prefer per-output hints?

Reply:
Yes we should do them per output. People will fix one and miss one and then be scratching their heads on why the linter is mad at them. We need to be specific.